src/main/java/com/hw/bookstore/domain/repository/BookRepositoryImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.hw.bookstore.domain.repository;
+
+import com.hw.bookstore.domain.entity.Book;
+import com.hw.bookstore.exception.DataProcessingException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@RequiredArgsConstructor
+@Repository
+public class BookRepositoryImpl implements BookRepository {

```

Comment:
Let's use JPA repository implementation ;)

Reply:
I see that you have switched to using JPARepository in the third HW, so you can ignore this comment